src/main/java/filter/SemanticException.java

```diff
@@ -0,0 +1,9 @@
+package filter;
+
+@SuppressWarnings("serial")
+public class SemanticException extends FilterException {
+    public SemanticException(String message) {
+        super("Semantic error in filter: " + message);
```

Comment:
I am already planning to change the message and add suggestion of supported inputs